server/src/test/java/org/spine3/server/validate/AbstractValidatingBuilderShould.java

```diff
@@ -0,0 +1,110 @@
+import static com.google.common.collect.Lists.newArrayList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.spine3.base.Identifiers.newUuid;
+import static org.spine3.base.Stringifiers.listStringifier;
+import static org.spine3.base.Types.listTypeOf;
+
+/**
+ * @author Illia Shepilov
+ */
+public class AbstractValidatingBuilderShould {
+
+    private AbstractValidatingBuilder<Task> validatingBuilder;
+
+    @Before
+    public void setUp() {
+        validatingBuilder = new ConcreteValidatingBuilder();
+    }
+
+    @Test
+    public void return_converted_value() throws ConversionException {
+        final Type type = listTypeOf(Integer.class);
+        final Stringifier<List<Integer>> stringifier = listStringifier(Integer.class);
+        StringifierRegistry.getInstance()
+                           .register(stringifier, type);
+        final List<Integer> convertedValue =
+                validatingBuilder.getConvertedValue(type, "\"1\"");
+        final List<Integer> expectedList = newArrayList(1);
+        assertThat(convertedValue, is(expectedList));
+    }
+
+    @Test(expected = ConversionException.class)
+    public void throw_exception_when_appropriate_stringifier_is_not_found() throws
+                                                                            ConversionException {
+        final String stringToConvert = "{value:1}";
+        validatingBuilder.getConvertedValue(TaskId.class, stringToConvert);
+    }
+
+    @Test(expected = ConversionException.class)
+    public void throw_exception_when_string_cannot_be_converted() throws ConversionException {
+        final String stringToConvert = "";
+        final Type type = listTypeOf(Timestamp.class);
+        final Stringifier<List<Timestamp>> stringifier = listStringifier(Timestamp.class);
+        StringifierRegistry.getInstance()
+                           .register(stringifier, type);
+        validatingBuilder.getConvertedValue(type, stringToConvert);
+    }
+
+    @Test
+    public void validate_value() throws ConstraintViolationThrowable {
+        final FieldDescriptor descriptor = ProjectId.getDescriptor()
+                                                    .getFields()
+                                                    .get(0);
+        validatingBuilder.validate(descriptor, newUuid(), "id");
+    }
+
+    @Test(expected = ConstraintViolationThrowable.class)
+    public void throw_exception_when_field_contains_constraint_violations()
+            throws ConstraintViolationThrowable {
+        final FieldDescriptor descriptor = PatternStringFieldValue.getDescriptor()
+                                                                  .getFields()
+                                                                  .get(0);
+        validatingBuilder.validate(descriptor, "incorrectEmail", "email");
+    }
+
+    private static class ConcreteValidatingBuilder extends AbstractValidatingBuilder<Task> {
```

Comment:
Please rename it so that you don't reference any [construction material stuff](https://en.wikipedia.org/wiki/Concrete).